mongo/integration/crud_prose_test.go

```diff
@@ -113,6 +114,136 @@ func TestWriteErrorsWithLabels(t *testing.T) {
 
 }
 
+func TestWriteErrorsDetails(t *testing.T) {
```

Comment:
While it tests the same behavior, the test here doesn't check all the things that the [test](https://github.com/mongodb/specifications/tree/master/source/crud/tests#writeerror-details-exposes-writeerrors-errinfo) described in the crud spec asks for.